### DISCUSSION TITLE
Water Valve

### DISCUSSION THREAD

**Discussion by enunes19** [dgreif/ring#1392] [Ideas]
Thinking it might be a good feature to add water valves into Homebridge from Ring. They just updated Ring Alarm to work with the Zooz Titan a few days ago and it can be controlled from the flood freeze sensors.

**Comment by tsightler** [comment]
Do you own one of these?  Have you connected it to your Ring Alarm?  Does it not already work?  I'd be surprised if it shows up as anything other than a switch as that how it appears to work when it's connected via Z-wave directly to HA and I'm not sure if there's any other category.  Regardless, simply posting here isn't the way to get a device added, someone will need to actually setup a device with the Ring Alarm and provide the [data discovery](https://github.com/dgreif/ring/wiki/Data-Discovery) and provide it to an open issue, then there is at least a chance it could be added.

**Reply by enunes19** [reply]
Sorry for the late reply. I do have one. It used to integrate with Ring Alarm as a switch, but Ring recently added full support so it integrates with the flood/freeze sensors. When it was a switch in Ring it also showed as a switch in Homebridge. After Ring added full support, it now shows up in Homebridge as a water valve, but is hidden. I haven't been able to find a way to expose it as an active accessory. I would be willing to provide the data discovery. I was hoping it could show under the Water summaries in the Home app if it could be integrated with Homebridge.

**Reply by tsightler** [reply]
It already shows up in Homebridge as a water valve?  That's quite surprising as I'm not sure how that would even be possible, but, if it does, it makes me wonder if there's anything that needs to be done in the plugin at all.  Please provide the data discovery and we can look at it.

**Reply by enunes19** [reply]
Yes here is a screenshot of what shows in Homebridge under the logs. I believe it is showing as a valve, but I am also new to this and still learning some of the more advanced things in Homebridge so I may be wrong.  

<img width="942" alt="Screenshot 2024-02-03 at 6 07 45 PM" src="https://github.com/dgreif/ring/assets/157775082/427a0656-4145-40ea-ace9-c98afd5a6eca">

The data discovery is too long to fit into a post. What's the best way to make sure you have it? Thank you again for your help!

**Reply by tsightler** [reply]
Ah, I see, you just meant in the logs. It's "hidden" because there no support for device type of valve.water.

> The data discovery is too long to fit into a post. What's the best way to make sure you have it?

This is why I stated earlier that adding additional device support should normally be opened as an issue vs a discussion, however, you should be able to attack the file here, zipped or whatever.  However, if you prefer, you can email it to me using my same username at gmail.

**Comment by enunes19** [comment]
Great thank you!

**Reply by tsightler** [reply]
Thanks for providing the device data, looks pretty straightforward, state is "open" or "closed" and it looks like it uses some custom commands to open/close the valve, very similar implementation to a Z-wave lock via Ring.

**Reply by enunes19** [reply]
Thank you for looking into it! I am glad it seems pretty straightforward.

**Reply by vandreykiv** [reply]
Hello @tsightler,

Do you have any updates? Maybe I can help somehow? I do have titan and see that it is discovered but hidden in the logs. 

Thanks,
Slava
